### PR TITLE
Issue 1741 snap

### PR DESCRIPTION
**Pull request by FilipLeitner**

### CODE REVIEW

File: projects/hslayers/src/assets/locales/en.json [hunk 1]
```diff
@@ -376,7 +376,9 @@
     "saveDrawingToLayer": "Save drawing to layer",
     "Select layer": "Select layer",
     "serverLayers": "Server layers",
-    "unsavedDrawing": "Unsaved drawing"
+    "unsavedDrawing": "Unsaved drawing",
+    "allowSnapping": "Allow snapping",
+    "snapTo": "Snap to features of layer:"
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/en.json:381]: I would leave the colon out of the translation string. For translating convenience.

File: projects/hslayers/src/components/draw/draw.service.ts [hunk 2]
```diff
@@ -52,6 +52,13 @@ export class HsDrawService {
   hasSomeDrawables: boolean;
   draw: Draw;
   modify: any;
+
+  //Snap interaction
+  snap: Snap;
+  snapActive = false;
+  snapSource: any;
```
Comment (raitisbe) [inline projects/hslayers/src/components/draw/draw.service.ts:59]: VectorLayer

File: projects/hslayers/src/assets/locales/en.json (hunk 1 above)
Comment (FilipLeitner) [reply projects/hslayers/src/assets/locales/en.json]: e7301c4

File: projects/hslayers/src/components/draw/draw.service.ts (hunk 2 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/draw/draw.service.ts]: d9db600

**Comment by FilipLeitner** [comment]
Should also note that for now its only available in draw panel. I feel like toolbar is already filled up with various functionality  and  adding another would make it bit opaque. So I was a kinda reluctant to add it there. What do you think @raitisbe  ?

File: projects/hslayers/src/components/draw/draw.service.ts [hunk 3]
```diff
@@ -612,6 +640,50 @@ export class HsDrawService {
...
+
+  /**
+   * Handle snap interaction changes
+   * Remove snap interaction if it already exists, recreate it if source is provided.
+   */
+  toggleSnapping(source?: VectorSource): void {
+    this.HsMapService.loaded().then((map) => {
+      this.snapSource = source ? source : this.snapSource;
+      if (this.snapActive && this.snapSource) {
+        if (this.snap) {
```
Comment (raitisbe) [inline projects/hslayers/src/components/draw/draw.service.ts:660]: There seems to be unnecessary duplication

File: projects/hslayers/src/components/draw/draw.service.ts [hunk 4]
```diff
@@ -612,6 +640,50 @@ export class HsDrawService {
...
+        }
+      }
+    });
+  }
+  /**
+   * Changes layer source of snap interaction
+   */
+  changeSnapSource(layer: Layer): void {
+    const isLayerClustered = this.HsLayerUtilsService.isLayerClustered(layer);
+    let snapSourceToBeUsed;
```
Comment (raitisbe) [inline projects/hslayers/src/components/draw/draw.service.ts:680]: Ternary operator?

Comment (FilipLeitner) [reply projects/hslayers/src/components/draw/draw.service.ts:680]: Sure, itll be cleaner. I just coppied this part from else where so I might as well refactor both occurences.

Comment (raitisbe) [reply projects/hslayers/src/components/draw/draw.service.ts]: great, thanks

**Review by raitisbe — APPROVED** [review]

**Pull request merged**